### ISSUE TITLE
cannot upgrade - unzips to %AppData%\Local\Temp\tablacus

### ISSUE THREAD

**Issue by AlanJayWeiner** [tablacus/TablacusExplorer#260]
Apparently I have not been able to upgrade for quite a while - I'm still running 19.3.17.   It's been running the 'install upgrade', but it hasn't actually performed the upgrade itself.

Tablacus is installed to "C:\Portable Apps\Tablacus"
The config settings are in "C:\Portable Apps\Tablacus\config"
(both paths were copied from the About Tablacus Explorer dialog)
Information:   TE64 19.3.17 Win 10.0.18363 WS 300 IE 11 en_us 240 Windows Defender
Add-ons:   mouse,key,addressbar,filterbar,up,treeview,toolbar,mainmenu,titlebar,tabplus,multithread,remember,undoclosetab,inneraddressbar,framepane,addtabs2fav,foldermenusettings,totalfilesizesort,history,statusbar,controlpanelbutton,extensioncolor,tabs,tabgroups,startup,sendto,savecustcolors,recentlyclosedtabs,innerback,innerforward,autoupdate,countbar,fullpathbar,font,tabname,lockedtabstop,switchpane,teracopy,clipboardhistory,labelgroups,tabparentname,favoritesnewtab,grabbar,sizestatus,addonsupdater,branchbar,color,innerup,autosave,refresh,split,addonswitcher,favorites,colorlabels,innernewfolder,labelbutton,newtab,hotbutton,label,dragdrop,folderlistmenu,split3,split6,tabcolorplus,tabdesign,underline,inactivepane,autobackup

When I try to upgrade:  it unzips to %AppData%\Local\Temp\tablacus, Tablacus closes, but it does not restart, and does not upgrade the actual running copy (C:\Portable Apps\Tablacus\TE64.exe)

I may have messed up a setting somewhere - at one point I tried the Chocolatey version of Tablacus (and later uninstalled it).   I thought there was a PortableApps.com platform version that I had tried also, but I don't see that on their website anymore; I may just be remembering wrong.  (I do like and run numerous PortableApps.com portable apps - really like them!)

I searched the registry to see if anything pointed to "temp\tablacus" but didn't find anything.   I did find numerous entries pointing to "C:\Portable Apps\Tablacus\TE64.exe"  (and TE32.exe)

As you can see, I have a lot of add-ons.  I also have about 12 groups, and probably 70+ tabs between all the groups.   If possible, I'd like to avoid a new installation - I don't want to lose the current configuration and all my groups and tabs.
I can send you any xml config files if they'll help.

Thanks for writing Tablacus!   It can be confusing and complicated, but it's incredible - I've recommended it to many other people.

And - Happy New Year!   (sorry to be opening the new year with a problem report!)

**Comment by tablacus** [comment]
Could you please try the following?

1. Copy folder: "C:\Portable Apps\Tablacus" to "C:\Portable Apps\Tablacus2".
2. Run "C:\Portable Apps\Tablacus2\te64.exe"
3. Update: Help -> Check for updates.

If the above method doesn't work,

1. Copy folder: "C:\Portable Apps\Tablacus" to "C:\Portable Apps\Tablacus3".
2. Download [te191222.zip](https://github.com/tablacus/TablacusExplorer/releases/download/19.12.22/te191222.zip).
3. Unzip te191222.zip to "C:\Portable Apps\Tablacus3".
4. Run "C:\Portable Apps\Tablacus3\te64.exe"
5. Check version: Help -> About Tablacus Explorer.

If the above method doesn't work,

1. Create folder: "C:\Portable Apps\Tablacus4".
2. Download [te191222.zip](https://github.com/tablacus/TablacusExplorer/releases/download/19.12.22/te191222.zip).
3. Unzip te191222.zip to "C:\Portable Apps\Tablacus4".
4. Copy folder "C:\Portable Apps\Tablacus\addons" to "C:\Portable Apps\Tablacus4\addons".
5. Copy folder "C:\Portable Apps\Tablacus\config" to "C:\Portable Apps\Tablacus4\config".
6. Run "C:\Portable Apps\Tablacus4\te64.exe"
7. Check version: Help -> About Tablacus Explorer.

Happy New Year to you too!

**Comment by AlanJayWeiner** [comment]
Hi Gaku,
Finally got time to work with this; sorry for the delay.

"Tablacus2" had the same result I've been seeing; no surprise there.

"Tablacus3" and "Tablacus4" initially work, but as soon as I update the add-ons, they fail.
I had add-ons updater enabled, so about 5 seconds after "check for updates" it updated the addons.
After a few seconds, I get an error dialog:

![Screenshot 2020-01-04 17 23 16 - tablacus error dialog](https://user-images.githubusercontent.com/25418440/71772865-f68b3a80-2f20-11ea-9aa4-bae82f6b07d7.png)

Selecting either "yes" or "no" ends up the same - all my groups get removed except for the first group, and it is renamed "Group1"
I was in a different group; it kept the *first* group; not the active group.

I tried several experiments with "Tablacus4".   copying \addons gives several conflicts.  I tried both ways; keeping my old version, and keeping the new version.   Same results - when it updates the addons I lose all my groups.
(each experiment started with a fresh copy, of course)

I think I tracked it down to the "Tab Groups" add-on.
I did "Tablacus4" steps as above.
It doesn't matter about the conflicts in \addons (keeping old, or using new).  Both ways fail the same.
I disabled "Add-ons updater" so nothing updated in background.
I did Tools -> Add-ons, and selected "Get Add-ons".  It showed lots of updates available.
I can update all the add-ons *except* Tab Groups, and it keeps my groups and tabs.
As soon as I update Tab Groups, I lose all my groups except the first one.

I get the error dialog about line 1267 in common.js every time I do Tools -> Add-ons and select "Ok", even after updating Tab Groups and losing all my groups.   Selecting "Cancel" does not give an error.

If you'd like, I can zip my "old" Tablacus and send it to you.  How can I send it?   (I can share it on Dropbox if that's easier - just let me know how to get the link to you)

**Comment by tablacus** [comment]
Please send link to gaku@tablacus.com

Thank you,

**Comment by tablacus** [comment]
I have received your file.
Let me check.

Thank you,

**Comment by AlanJayWeiner** [comment]
thanks!

**Comment by tablacus** [comment]
I have adjusted a problem with "Auto backup".
Please try to update it.

Regards,

**Comment by AlanJayWeiner** [comment]
Hi Gaku!
That worked!   I updated "Auto backup" first, then Check for updates.
Tablacus and addons updated fine, but it hung after updating ("Tablacus is
not responding").   I don't remember if it closed and reopened before it
hung (does it do that before updating addons?)  All the addons had been
updated (based on my backup while it was hung)

I made a backup before updating anything.
Updated Auto backup (I forget if I closed/reopened before Check for updates)
Check for updates.
Magic occurs...   Tablacus hung.   I made a backup while it was hung.
Forced it closed with Process Explorer - neglected to make a backup after
closing it (duh!).
Started it up; worked fine - made a backup while it was running.
Closed Tablacus, made another backup.
The only differences were in history.tsv, remember.xml, and the windows.xml
files.  (except for the backup before updating, of course)
If you want to look at any of those backups, I put them on Dropbox.

It's been running fine since the update now.

Thank you so much!  Domo arigato gozaimashita!

Tablacus is what I wanted for many, many years.  I didn't like any of the
other Explorer replacements for various reasons - I tend to be very picky
in my workflow...  :)
I've been using Tablacus for several years now, and would hate to live
without it.  (I'm *still* learning all the settings and addons and their
settings).
I've recommended it to many people over the years (with the warning that it
can be complicated...)

I especially like that it's a portable app (although I wish it were in
PortableApps.com ".paf" format...  that makes things so easy...)
I've been switching to portable apps since my last notebook broke; -
reinstalling and reconfiguring everything - I'd love never to have to do
that again.   (*and* portable apps don't clutter the registry much!)

Thank you again for developing it, and thank you for fixing my problem!

All the best,
Alan
